api/v1beta1/azureimage_validation_test.go

```diff
@@ -223,3 +223,65 @@ func createTestImageByID(imageID string) *Image {
 		ID: &imageID,
 	}
 }
+
+func TestValidateSingleDetailsOnly(t *testing.T) {
+	g := NewWithT(t)
+
+	// Case 1: image.Marketplace != nil, image details are found
```

Comment:
Thanks for the update! Now that I look at this again, do you think you could refactor this like [this test func](https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/858c9c5f3abf72152bf3860c2a2a26d81b646140/api/v1beta1/azureimage_validation_test.go#L130-L161) where we define a list of testcase structs and loop over them to check the result/error? I think it would make sense given that the test cases all have the same type of input and error.